polaris/hub/client.py

```diff
@@ -320,7 +320,7 @@ def open_zarr_file(
         polaris_fs = PolarisFileSystem(
             polaris_client=self,
             dataset_owner=owner,
-            dataset_name=name,
+            dataset_name=sluggify(name),
```

Comment:
**Suggestion:** This is a good catch! I would move it inside the `PolarisFileSystem` constructor, however, to centralize the logic.